upgrade_compatibility.md

```diff
@@ -0,0 +1,12 @@
+# Upgrade Compatibility
+
+While we try and make all changes backwards-compatible, sometimes we have to make changes that mean you cannot cleanly provision a new environment.
+
+
+## DNS resources already exist
+
+Error:
+
+`Error creating DNS RecordSet: googleapi: Error 409: The resource 'entity.change.additions[0]' named 'env-proxy.tsuru2.paas.alphagov.co.uk. (A)' already exists, alreadyExists`
+
+This is caused by `7780fe9d`, which also contains the scrips you can use to fix it.
```

Comment:
Maybe also link to PR (which in this case contains more detais & workarounds)? That commit ID is findable only in github log, but not on github web (not even in search)...